client/data/authorizations/actions.ts

```diff
@@ -39,3 +71,58 @@ export function updateAuthorizationsSummary(
 		query,
 	};
 }
+
+export function* submitCaptureAuthorization(
+	paymentIntentId: string,
+	orderId: number
+): Generator< unknown | Authorization > {
+	try {
+		yield dispatch( STORE_NAME, 'startResolution', 'getAuthorization', [
+			paymentIntentId,
+		] );
+
+		const result = yield apiFetch( {
+			path: `/wc/v3/payments/orders/${ orderId }/capture_authorization`,
+			method: 'post',
+			data: {
+				payment_intent_id: paymentIntentId,
+			},
+		} );
+
+		const authorization = {
+			payment_intent_id: ( result as any ).id,
+			captured: ( result as any ).status === 'succeeded',
+		};
+
+		yield updateAuthorization( authorization as Authorization );
+
+		// Need to invalidate the resolution so that the components will render again.
+		yield dispatch(
+			STORE_NAME,
+			'invalidateResolutionForStoreSelector',
+			'getAuthorizations'
+		);
+
+		yield dispatch(
+			STORE_NAME,
+			'invalidateResolutionForStoreSelector',
+			'getAuthorizationsSummary'
+		);
+		// Create success notice.
+		yield dispatch(
+			'core/notices',
+			'createSuccessNotice',
+			__( 'You have captured the payment.', 'woocommerce-payments' )
+		);
+	} catch ( error ) {
```

Comment:
The catched `error` isn't used here and thus not exposed to the user. Should we include it part of the message/notice to give the user some context?

Reply:
I think this is a good point, but I checked other places where we use the createErrorNotice, and we usually add a generic message, without using the catched error message. Maybe this is on purpose, to avoid exposing too much information. I don't know what to do here to be honest.

Reply:
I agree. I'm just thinking of catching the known errors already documented here #4957 

```
 -   `wcpay_missing_order` - Order not found
 -   `wcpay_refunded_order_uncapturable` -  Payment cannot be captured for partially or fully refunded orders
 -   `wcpay_payment_uncapturable` - The payment cannot be captured if intent status is not one of 'processing', 'requires_capture', or 'succeeded'
 -   `wcpay_intent_order_mismatch` - Payment cannot be captured because the order id does not match
 -   `wcpay_capture_error` - Unknown error
 ```

I think as a customer I would love to see `You can not capture this payment as it is partially/fully refunded` instead of `There has been an error capturing the payment. Please try again later.`. Because I'll keep trying later multiple times, but it will never work 😅 

I think having generic errors will just flood support with multiple requests that can be avoided

Reply:
+1 to @elazzabi's point. I had a similar concern in my error handling code for get (single) authorization:

https://github.com/Automattic/woocommerce-payments/pull/4688/files#diff-20dbcc4a43fe9db63759c8f939bed0803e2cbecdd715f5f6badc5d27189c09b1R56

The get auth endpoint is pretty simple and doesn't return a host of error codes, so there was not much need for the error object there. Here is a different story. I realize that would be additional work, but it'd definitely set a good example for others to follow in future.

Reply:
Thanks for the suggestion @anu-rock @elazzabi. I think this is a valid point and I will look into it. I might create a separate issue to tackle it after we solve the review bottleneck but before going live. Would that work for you if I finally create it?

Reply:
I finally decided to create an issue to improve this https://github.com/Automattic/woocommerce-payments/issues/5002

Reply:
@mgascam Thank you! A separate issue is even better :)